actors/power/tests/harness/mod.rs

```diff
@@ -171,6 +198,90 @@ impl Harness {
     pub fn check_state(&self) {
         // TODO: https://github.com/filecoin-project/builtin-actors/issues/44
     }
+
+    pub fn update_pledge_total(&self, rt: &mut MockRuntime, miner: Address, delta: &TokenAmount) {
+        let st: State = rt.get_state();
+        let prev = st.total_pledge_collateral;
+
+        rt.set_caller(*MINER_ACTOR_CODE_ID, miner);
+        rt.expect_validate_caller_type(vec![*MINER_ACTOR_CODE_ID]);
+        rt.call::<PowerActor>(
+            Method::UpdatePledgeTotal as MethodNum,
+            &RawBytes::serialize(BigIntDe(delta.clone())).unwrap(),
+        )
+        .unwrap();
+        rt.verify();
+
+        let st: State = rt.get_state();
+        assert_eq!(prev + delta, st.total_pledge_collateral);
+    }
+
+    pub fn current_power_total(&self, rt: &mut MockRuntime) -> CurrentTotalPowerReturn {
+        rt.expect_validate_caller_any();
+        let ret: CurrentTotalPowerReturn = rt
+            .call::<PowerActor>(Method::CurrentTotalPower as u64, &RawBytes::default())
+            .unwrap()
+            .deserialize()
+            .unwrap();
+        rt.verify();
+        ret
+    }
+
+    pub fn update_claimed_power(
+        &self,
+        rt: &mut MockRuntime,
+        miner: Address,
+        raw_delta: &StoragePower,
+        qa_delta: &StoragePower,
+    ) {
+        let prev_cl = self.get_claim(rt, &miner).unwrap();
+
+        let params = UpdateClaimedPowerParams {
+            raw_byte_delta: raw_delta.clone(),
+            quality_adjusted_delta: qa_delta.clone(),
+        };
+        rt.set_caller(*MINER_ACTOR_CODE_ID, miner);
+        rt.expect_validate_caller_type(vec![*MINER_ACTOR_CODE_ID]);
+        rt.call::<PowerActor>(
+            Method::UpdateClaimedPower as MethodNum,
+            &RawBytes::serialize(params).unwrap(),
+        )
+        .unwrap();
+        rt.verify();
+
+        let cl = self.get_claim(rt, &miner).unwrap();
+        let expected_raw = &prev_cl.raw_byte_power + raw_delta;
+        let expected_adjusted = &prev_cl.quality_adj_power + qa_delta;
+        if expected_raw.is_zero() {
```

Comment:
Unless I'm missing something the go test overcomplicates this.  These two if statements can be replaced with `assert!(expected_raw, cl.raw_byte_power)` and `assert!(expected_adjusted, cl.quality_adj_power)`

Reply:
Sounds right. I'll leave the refactoring to another PR, though.